plugins/per-directory-history/per-directory-history.zsh

```diff
@@ -68,14 +68,14 @@ function per-directory-history-toggle-history() {
   if [[ $_per_directory_history_is_global == true ]]; then
     _per-directory-history-set-directory-history
     _per_directory_history_is_global=false
-    print -n "\nusing local history"
+    zle -I
+    echo "using local history"
   else
     _per-directory-history-set-global-history
     _per_directory_history_is_global=true
-    print -n "\nusing global history"
+    zle -I
+    echo "using global history"
   fi
-  zle .push-line
```

Comment:
I believe I removed these lines because I first tried using `zle -M "using local history"` and if you keep these lines, the message will not show up and the prompt will be regenerated as you describe in https://github.com/ohmyzsh/ohmyzsh/issues/12068. When using `zle -M` instead of `zle -I`, the prompt should not be regenerated.